File: src/a1twopointerpalindrome/Python/palindrome_checker.py
# palindrome_checker.py

def is_palindrome(s):
    left, right = 0, len(s) - 1

    while left < right:
        print(f"Current elements left index: '{
              s[left]}' and right index: '{s[right]}'")

        if s[left] != s[right]:
            print("The elements aren't the same, hence we return False")
            return False

        left += 1       # Heading towards the right
        right -= 1      # Heading towards the left

    # We reached the middle of the string without finding a mismatch, so it is a palindrome.
    return True

File: src/a1twopointerpalindrome/TS/palindromeChecker.ts
// palindromeChecker.ts

export function isPalindromeTS(s: string): boolean {
    let left = 0;
    let right = s.length - 1;

    while (left < right) {
        console.log(`Current elements left index: '${s[left]}' and right index: '${s[right]}'`);

        if (s[left] !== s[right]) { // If the elements at index l and index r are not equal,
            console.log("The elements aren't the same, hence we return false");
            return false;
        }

        left++;         // Heading towards the right
        right--;        // Heading towards the left
    }

    return true;        // We reached the middle of the string without finding a mismatch, so it is a palindrome.
}
